### ISSUE TITLE
pythonic documentation for subset

### ISSUE THREAD

**Issue by pdknsk** [fonttools/fonttools#826]
```Python
from fontTools import subset
help(subset)
```
That lists the command line options for ```pyftsubset```, which is apparently what Google uses, but it's of course not necessary to go the ```subprocess``` route to subset fonts. Right now, to use ```subset``` directly, you have to ```dir``` your way to the correct usage, or read the source, as ```help``` is of little help.
```Python
options = subset.Options() # dir(options)
font = subset.load_font('font.ttf', options)
subsetter = subset.Subsetter(options)
subsetter.populate(text = 'Google')
subsetter.subset(font)
options.flavor = 'woff'
subset.save_font(font, 'font.woff', options)
```

**Comment by anthrotype** [comment]
Sorry but... what was the question?

**Comment by pdknsk** [comment]
​There isn't any. It's a request.​

**Comment by anthrotype** [comment]
`$ pyftsubset --help`

**Comment by pdknsk** [comment]
Yes, that's the bug. The lack of pythonic documentation, which is the request.

**Comment by anthrotype** [comment]
Well, it wasn't that clear from your message to be honest.
Sorry if I dismissed this too quickly.
Yes, we have very sparse documentation for the usage of fonttools as a library and currently most people just rely on the source code.
You are very welcome to send a pull request to improve this situation.
It's just not at the very top of my priorities at the moment, but I understand there's need for this.

**Comment by pdknsk** [comment]
Yes, my comment might've been a bit sparse itself.

I guess mentioning `subprocess` added some confusion. What I meant is that from within Python you obviously don't need to use `subprocess` to call `pyftsubset` but can instead use the library directly.

The code posted is just an example on how to use it.

**Comment by anthrotype** [comment]
Yes, that's a good start.
Thanks

**Comment by behdad** [comment]
> That lists the command line options for pyftsubset, which is apparently what Google uses

Not that it matters, but that's not true.  We call the subsetter through code.  pyftsubset was added mostly for font developers...
